php/route_guide/README.md

```diff
@@ -0,0 +1,259 @@
+```
+
+As you can see, we create and populate a request object, i.e. an `examples\Point` object. Then, we call the method on the stub, passing it the request object. If there is no error, then we can read the response information from the server from our response object, i.e. an `examples\Feature` object.
+
+```php
+  print sprintf("Found %s \n  at %f, %f\n", $feature->getName(),
+                $feature->getLocation()->getLatitude() / COORD_FACTOR,
+                $feature->getLocation()->getLongitude() / COORD_FACTOR);
+```
+
+#### Streaming RPCs
+
+Now let's look at our streaming methods. Here's where we call the server-side streaming method `ListFeatures`, which returns a stream of geographical `Feature`s:
+
+```php
+  $lo_point = new examples\Point();
+  $hi_point = new examples\Point();
+
+  $lo_point->setLatitude(400000000);
+  $lo_point->setLongitude(-750000000);
+  $hi_point->setLatitude(420000000);
+  $hi_point->setLongitude(-730000000);
+
+  $rectangle = new examples\Rectangle();
+  $rectangle->setLo($lo_point);
+  $rectangle->setHi($hi_point);
+
+  $call = $client->ListFeatures($rectangle);
+  // an iterator over the server streaming responses
+  $features = $call->responses();
+  foreach ($features as $feature) {
+    // process each feature
+  } // the loop will end when the server indicates there is no more responses to be sent.
+```
+
+The `$call->responses()` method call returns an iterator. When the server sends a response, a `$feature` object will be returned in the `foreach` loop, until the server indiciates that there will be no more responses to be sent.
+
+The client-side streaming method `RecordRoute` is similar, except there we pass the method an iterator and get back a `examples\RouteSummary`.
+
+```php
+  $points_iter = function($db) {
+    for ($i = 0; $i < $num_points; $i++) {
+      $point = new examples\Point();
+      $point->setLatitude($lat);
+      $point->setLongitude($long);
+      yield $point;
+    }
+  };
+  // $points_iter is an iterator simulating client streaming
+  list($route_summary, $status) =
+    $client->RecordRoute($points_iter($db))->wait();
+```
+
+Finally, let's look at our bidirectional streaming RPC `routeChat()`. In this case, we just pass a context to the method and get back a `BidiStreamingCall` stream object, which we can use to both write and read messages.
+
+```php
+$call = $client->RouteChat();
+```
+
+To write messages from the client:
+
+```php
+  foreach ($notes as $n) {
+    $route_note = new examples\RouteNote();
+    $call->write($route_note);
+  }
+  $call->writesDone();
+```
+
+To read messages from the server:
+
+```php
+  while ($route_note_reply = $call->read()) {
+    // process $route_note_reply
+  }
+```
+
+Each side will always get the other's messages in the order they were written, both the client and server can read and write in any order — the streams operate completely independently.
+
+<a name="try"></a>
```

Comment:
I think I might move this section (Try it out) up before Defining a Service, as in the Objective-C one, and add:
"The next sections guide you step-by-step through how this proto service is defined, how to generate a client library from it, and how to create an app that uses that library."
 I think the structure needs to be different from the examples where you're actually showing how to define and create a server - otherwise it seems a bit odd to be defining a service and then implementing a client for it without there being any mention of a server until the end.